frontend/src/components/module-editor/SideBarModuleOverview.tsx

```diff
@@ -9,15 +9,24 @@ import { Modal } from "../common/Modal";
 import { TextInput } from "../common/TextInput";
 
 import LessonItem from "./LessonItem";
+import DeleteModal from "../common/DeleteModal";
+
+enum ModalType {
+  EDIT = "edit",
+  DELETE = "delete",
+  CREATE_LESSON = "create-lesson",
+}
 
 const SideBarModuleOverview = (): React.ReactElement => {
   const context = useContext(EditorContext);
   const { isOpen, onOpen, onClose } = useDisclosure();
```

Comment:
Instead of adding a `ModalType` enum, it would be better to have three `useDisclosure` calls, that way the modal components don't have to mount/unmount when the type changes.